src/api/app/models/event/request.rb

```diff
@@ -134,8 +134,8 @@ def source_watchers
 
     def find_watchers(project_key)
       project_names = payload['actions'].map { |action| action[project_key] }.uniq
-      projects = Project.where(name: project_names).joins(watched_projects: :user)
-      projects.flat_map { |project| project.watched_projects.map(&:user) }
+      watched_projects = WatchedProject.where(project: Project.where(name: project_names))
+      User.where(id: watched_projects.select(:user_id))
```

Comment:
before we where returning an array, while now a relations is returned. Maybe we want to keep returning an array using `find` instead:

`User.find(watched_projects.select(:user_id))`

I also find it more readable. I do not know if using `find` instead of `where` can affect performance (either in a negative or a positive way).

Reply:
IMO returning a relation instead of an array is an improvement since you can always call `.to_a` on a relataion but you cant really turn an array into a relation.

Reply:
it doesn't matter - the next call on the relation is .each - which will cast into an array. But when you return a relation the caller has the freedom to use in_batches.

Reply:
ok, then let's merge this! 😉